conda_forge_tick/models/common.py

```diff
@@ -122,14 +146,38 @@ def none_to_empty_dict(value: T | None) -> T | dict[Never, Never]:
 GitUrl = Annotated[Url, UrlConstraints(allowed_schemes=["git"])]
 
 
-class LazyJsonReference(StrictBaseModel):
+VersionString = Annotated[str, Field(..., pattern=r"^\d+(\.\d+){0,2}$")]
+"""
+A string that matches version numbers in one of the following formats:
+- `X`
+- `X.Y`
+- `X.Y.Z`
+"""
```

Comment:
This regex is too restrictive in terms of allowed version strings.

Reply:
Currently, this field is only used for the field `smithy_version`. Is there any other schema that conda-smithy follows for its version numbers?

Reply:
Not right now yes but we could change smithy in the future. Restricting to just smithy's current semver scheme adds extra coupling between this code and smithy. 

If we are going to go to the trouble of validating that the version string from conda smithy is correct, then we should definitely use a regex that covers all of the possible allowed versions /. Schemes allowed by conda.  

On the other hand, validating that smithy has a correctly formatted version is adding several lines of code to catch an error that is unlikely to ever happen. From that point of view, I'd be just as happy to remove this bit of schema validation.

Reply:
Ah I see. Thanks for finding this!

I'd check the conda source code for a version regex or maybe google around a bit.

Reply:
@beckermr I decided to use `VersionOrder` because there seems to be a lot of custom logic involved. Think it's ne cleanest solution.

Should be fixed with this.